keras_nlp/models/whisper/whisper_audio_feature_extractor.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+
+NUM_MELS = 80
```

Comment:
I'm looking more through the code you linked earlier about needing to support `NUM_MELS=80` always. It looks like originally they need to do that because they are keeping the "mel_filters" in a separate file asset. https://github.com/openai/whisper/blob/main/whisper/audio.py#L83

Now that we are computing these filters outselves on the fly, can we just remove the restriction? And leave num_mels as configurable on this extractor and backbone (and only support 80 in our presets)?

That would be less jarring than needing to keep referencing this constant, or have a parameter with only one possible value (which is awkward).